src/components/donation/Steps/DonateMethods/Stocks/Form.tsx

```diff
@@ -0,0 +1,98 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Field } from "components/form";
+import { FormProvider, UseFormReturn, useForm } from "react-hook-form";
+import { useDispatch } from "react-redux";
+import { StockFormStep, setDetails } from "slices/donation";
+import { object, string } from "yup";
+
+export default function Form(props: StockFormStep) {
+  const dispatch = useDispatch();
+
+  const methods = useForm({
+    defaultValues: props.details
+      ? {
+          symbol: props.details.symbol,
+          numShares: props.details.numShares.toString(),
+        }
+      : {
+          symbol: "",
+          numShares: "",
+        },
+    resolver: yupResolver(
+      object({
+        symbol: string().required("required"),
+        numShares: string()
+          .required("required")
+          .matches(/^[1-9]\d*$/, "invalid"),
+      })
+    ),
+  });
+
+  type FV = typeof methods extends UseFormReturn<infer U> ? U : never;
+
+  return (
+    <FormProvider {...methods}>
+      <form
+        className="grid"
+        onSubmit={methods.handleSubmit((fv) => {
+          dispatch(
+            setDetails({
+              method: "stocks",
+              symbol: fv.symbol,
```

Comment:
```suggestion
              symbol: fv.symbol.toUpperCase(),
```
nit: **the PR can be merged as this is not a blocker**, but ticker symbols are usually upper case. @SovereignAndrey this is more of decision for the "product owner" (you), so tagging you to see the message.

I entered `aapl` into the form and got the generated email that kept this lowercase, although I expected it to uppercase it for me:
![image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/19427053/1cca3662-acb3-4733-8b0f-3402bd5faf06)